appcues/src/main/java/com/appcues/ui/primitive/TextInputPrimitive.kt

```diff
@@ -42,11 +40,7 @@ import java.util.UUID
 internal fun TextInputPrimitive.Compose(modifier: Modifier) {
 
     val formState = LocalExperienceStepFormStateDelegate.current
-    val text = remember { mutableStateOf(defaultValue ?: "") }
-
-    LaunchedEffect(key1 = text.value) {
-        formState.captureFormItem(this@Compose.id, SingleTextFormItem(label.text, required, text.value))
-    }
+    val text = remember { mutableStateOf(formState.getValue(this)) }
```

Comment:
the value now comes from the form state, so it is restored when you return to this composition from elsewhere, if a value was already set.